src/wallet/index.js

```diff
@@ -0,0 +1,414 @@
+        satoshis: changeOutputAmount
+      })
+      transaction = transaction.addOutput(output)
+    }
+    // NOTE: We are not using Bitcore to set change
+
+    // Shift any remainder to other outputs randomly
+    let outputIndex = -1
+    const mutableOutputs = exactOutputs ? transaction.outputs.length - outputs.length : transaction.outputs.length
+    // NOTE: we don't change the number of outputs here, but we also don't want to execute anythign if there are no outputs
+    // eslint-disable-next-line no-unmodified-loop-condition
+    while (mutableOutputs > 0) {
+      outputIndex = (outputIndex + 1) % mutableOutputs
+      const output = transaction.outputs[outputIndex + (exactOutputs ? outputs.length : 0)]
+      const delta = transaction.inputAmount - transaction.outputAmount
+      const finalSize = transaction._estimateSize() + transaction.outputs.length // Numbers are variable length in bitcoin, changing the output amount could add a byte to any given txn.
+      const properFee = Math.ceil(finalSize * feePerByte)
+      const upperBound = delta - properFee
+      const splitPosition = Math.floor(Math.random() * upperBound)
+      // Add a small amount to the current output
+      const amountToAdd = splitPosition > upperBound - splitPosition ? upperBound - splitPosition : splitPosition
+      console.log('Amount available', delta, 'Fee required', properFee, 'Will add', amountToAdd, 'To output #', outputIndex)
+      if (amountToAdd === 0) {
+        // We can't add anymore funds to any outputs
+        break
+      }
+      output.satoshis += amountToAdd
+      transaction._outputAmount += amountToAdd
+    }
+
+    // Sign transaction
+    transaction = transaction.sign(signingKeys)
+    const finalTxnSize = transaction._estimateSize()
+    // Sweep change into a randomly provided output.  Helps provide noise and obsfuscation
+    console.log('size', finalTxnSize, 'outputAmount', transaction.outputAmount, 'inputAmount', transaction.inputAmount, 'delta', transaction.inputAmount - transaction.outputAmount, 'feePerByte', (transaction.inputAmount - transaction.outputAmount) / transaction._estimateSize())
+
+    console.log(transaction)
+    return { transaction, usedIDs: usedUtxos.map(utxo => calcId(utxo)) }
+  }
+  get feePerByte () {
+    return 2
+  }
+  getAddressByElectrumScriptHash (scriptHash) {
+    return this.electrumScriptHashes[scriptHash]
+  }
+  getPaymentAddress (seqNum) {
+    return Object.keys(this.addresses)[seqNum]
+  }
+  get balance () {
+    return Object.values(this.storage.getUTXOs()).reduce((acc, output) => acc + output.satoshis, 0)
+  }
+  get myAddress () {
+    // TODO: Not just testnet
+    return this.identityPrivKey.toAddress('testnet')
+  }
+  get myAddressStr () {
+    // TODO: Not just testnet
+    return this.identityPrivKey.toAddress('testnet')
+      .toCashAddress()
+  }
+  get allAddresses () {
+    return { ...this.addresses, ...this.changeAddresses }
+  }
+  get privKey () {
```

Comment:
Perhaps `privKeys` rather than `privKey` given that it's an array of private keys?